pages/property-filter/async-loading.integ.page.tsx

```diff
@@ -51,31 +47,10 @@ export default function () {
     ],
     operation: 'and',
   });
-  const request = useRef<{
-    filteringText: string;
-    filteringProperty?: PropertyFilterProps.FilteringProperty;
-    filteringOperator?: PropertyFilterProps.ComparisonOperator;
-  }>();
-  const [filteringOptions, setFilteringOptions] = useState<PropertyFilterProps['filteringOptions']>([]);
-  const [status, setStatus] = useState<PropertyFilterProps['filteringStatusType']>('pending');
-  const fetchData = async (
-    filteringText: string,
-    filteringProperty?: PropertyFilterProps.FilteringProperty,
-    filteringOperator?: PropertyFilterProps.ComparisonOperator
-  ) => {
-    const items = await fetchOptions();
-    if (
-      !request.current ||
-      request.current.filteringText !== filteringText ||
-      request.current.filteringProperty !== filteringProperty ||
-      request.current.filteringOperator !== filteringOperator
-    ) {
-      // there is another request in progress, discard the result of this one
-      return;
-    }
-    setStatus('finished');
-    setFilteringOptions(items);
-  };
+  const [fetchTarget, setFetchTarget] = useState<'properties' | 'options'>('properties');
+  const propertiesLoader = useOptionsLoader({ timeout: 100 });
```

Comment:
it looks like we're passing filteringProperties as a constant anyway, does it play a role anywhere?

Reply:
Yes, that is to match the current implementation more or less. Extending the test page to also feature async loading for properties could be a good next step. It is used here:
```
const status = { properties: propertiesLoader, options: optionsLoader }[fetchTarget].status;
```

As you can see, the status is taken from the properties or options loader depending on the current fetch target.